backend/src/layer/trigger/scheduling/scheduler.rs

```diff
@@ -0,0 +1,251 @@
+use std::{sync::Arc, time::Instant};
+
+use crate::interface::{
+    image_review::ImageReviewScheduling, mealplan_management::MensaParseScheduling,
+};
+
+use tokio::sync::Notify;
+use tokio_cron_scheduler::{Job, JobScheduler};
+use tracing::{info, info_span, Instrument};
+
+/// Structure containing [cron](https://cron.help/)-like schedules for running actions regularly.
+///
+/// **Important:** Unlike regular cron expressions, seconds also have to be specified, see [here](https://lib.rs/crates/tokio-cron-scheduler).
+pub struct ScheduleInfo {
+    /// Cron-like schedule for running the image review process to check for no longer existing images, see [`ImageReviewScheduling`].
+    pub image_review_schedule: String,
+    /// Cron-like schedule for running the meal plan update process for the current day's meal plan, see [`MensaParseScheduling`].
+    pub update_parse_schedule: String,
+    /// Cron-like schedule for running the meal plan update process for all available meal plan data, see [`MensaParseScheduling`].
+    pub full_parse_schedule: String,
+}
+
+#[derive(PartialEq, Eq, Debug)]
+enum State {
+    Created,
+    Running,
+    Stopped,
+}
+
+/// Class fro planning regular events.
+pub struct Scheduler {
+    scheduler: JobScheduler,
+    state: State,
+}
+
+impl Scheduler {
+    /// Creates a new scheduler with time plans specified in `info` and actions specified in the `scheduling`s.
+    pub async fn new(
+        info: ScheduleInfo,
+        image_scheduling: impl ImageReviewScheduling + 'static,
+        parse_scheduling: impl MensaParseScheduling + 'static,
+    ) -> Self {
+        let scheduler = JobScheduler::new()
+            .await
+            .expect("cannot initialize scheduler");
+
+        // === image review ===
+
+        let image_review = Arc::new(image_scheduling);
+
+        let image_review_job =
+            Job::new_async(info.image_review_schedule.as_ref(), move |_, _| {
+                let image_review = image_review.clone();
+                Box::pin(async move {
+                    info!("Started image review.");
+                    let start = Instant::now();
+
+                    image_review.start_image_review().await;
+
+                    info!("Finished image review in {:?}.", start.elapsed());
+                }.instrument(info_span!("image_review")))
+            })
+            .expect("could not create schedule for image reviewing, you should also specify seconds in your cron expression");
+
+        scheduler
+            .add(image_review_job)
+            .await
+            .expect("could not add job for image reviewing to scheduler");
+
```

Comment:
.exspect errors should be logged in any way, to determine misbehavior

Reply:
When the program crashes by panicing, the panic message gets printed to the console automaticly. Shouldn't that be enough?